### PR TITLE
[BACKLOG-14028] - DET - Flat Table - Ordering Columns with Null Values

### PR DESCRIPTION
**Pull request by Aliaksandr-Kastenka**

### CODE REVIEW

**Comment by Aliaksandr-Kastenka** [comment]
@DFieldFL @kurtwalker 
Please review.

File: core/test-src/org/pentaho/di/core/row/value/ValueMetaBaseTest.java [hunk 1]
```diff
@@ -526,6 +526,11 @@ public void testCompareIntegers() throws KettleValueException {
     assertEquals( -1, intMeta.compare( int2, int1 ) );
     assertEquals( 0, intMeta.compare( int1, int1 ) );
 
+    int1 = null;
+    int2 = new Long( 6223372036854775804L );
+    intMeta.setSortedDescending( true );
+    assertEquals( 1, intMeta.compare( int1, int2 ) );
```
Comment (DFieldFL) [inline core/test-src/org/pentaho/di/core/row/value/ValueMetaBaseTest.java]: add another test for ascending please.

**Comment by Aliaksandr-Kastenka** [comment]
@DFieldFL 
Pull-request updated

**Comment by kurtwalker** [comment]
@Aliaksandr-Kastenka Can you point me to the code where the sort is initiated?  I'm wondering if we should pass in a comparator instead of putting the asc/desc property on the value meta object.

**Comment by Aliaksandr-Kastenka** [comment]
@kurtwalker In this case stacktrace is the following:

https://github.com/pentaho/pentaho-kettle/blob/master/core/src/org/pentaho/di/core/row/value/ValueMetaBase.java#L3499
https://github.com/pentaho/pentaho-kettle/blob/master/core/src/org/pentaho/di/core/row/RowMeta.java#L879
https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/org/pentaho/di/trans/steps/sort/SortRows.java#L708
...
https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/org/pentaho/di/trans/steps/sort/SortRows.java#L623

**Comment by kurtwalker** [comment]
thanks, I get it now.  makes sense.  :+1:

**Pull request merged**